src/ppx/scene/scene_material.cpp

```diff
@@ -183,22 +205,25 @@ scene::VertexAttributeFlags MaterialFactory::GetRequiredVertexAttributes(const s
     return attrFlags;
 }
 
-scene::MaterialRef MaterialFactory::CreateMaterial(
+scene::Material* MaterialFactory::CreateMaterial(
     const std::string& materialIdent) const
 {
-    scene::MaterialRef material;
+    scene::Material* pMaterial = nullptr;
 
     if (materialIdent == PPX_MATERIAL_IDENT_UNLIT) {
-        material = scene::MakeRef(new scene::UnlitMaterial());
+        pMaterial = new scene::UnlitMaterial();
     }
     else if (materialIdent == PPX_MATERIAL_IDENT_STANDARD) {
-        material = scene::MakeRef(new scene::StandardMaterial());
+        pMaterial = new scene::StandardMaterial();
+    }
+    else if (materialIdent == PPX_MATERIAL_IDENT_DEBUG) {
+        pMaterial = new scene::DebugMaterial();
     }
     else {
-        material = scene::MakeRef(new scene::ErrorMaterial());
+        pMaterial = new scene::ErrorMaterial();
```

Comment:
nit:
What about returning error if the value is `MATERIAL_IDENT_ERROR` and assert otherwise?
This way if we see an error material, we know it's an handled error, like something wrong in the GLTF scene, and not us who made a typo when using the fucntion?

Reply:
I think a warning is appropriate. Missing material is valid usage, so the loader shouldn't stop loading. The default ErrorMaterial renders flat purple to indicate something went wrong visually.

Reply:
The loader will log a warning if there's a missing material.